### PR TITLE
feat(65): coder deployment

### PR DESCRIPTION
**Pull request by benie-joy-possi**
## 1. Summary

This PR changes:

-  This PR introduces Coder as an App-of-Apps orchestrator across two repositories.     
-
It solves:

  - Re-introduces Coder as a self-hosted cloud dev environment platform                                                                                     
  - Follows established lightbridge App-of-Apps pattern (ADR-0019 canonical)                                                                                
  - Local k3s testing confirmed: Coder v2.33.7 + CNPG + Keycloak OIDC works end-to-end   
  - #651 

---

## 2. Intent

The intent of this PR is:

>  Re-introduce Coder (removed in ADR-0027) now that AI agent orchestration is a platform priority. Use the                                               mature lightbridge App-of-Apps pattern — no new architectural concepts. Deploy upstream Coder chart v2.33.7                                      via multi-source ArgoCD Application (wave 2) with CNPG database (wave 1), ExternalSecrets (wave 0), and per-                                            
  environment kustomize deps overlay. Certificate strategy: HTTP-01 ACME (no wildcard).

---

## 3. Scope

### In Scope

-  3 new Helm charts: orchestrator + secrets leaf + db leaf 
- Keycloak coder client scope, protocol mapper, roles  
- Production values + deps overlay in private ai-helm-values repo  
### Out of Scope

- Wildcard cert for *.coder-ai.camer.digital (HTTP-01 ACME used instead — path-based workspace apps)   
-  Workspace template configuration (Terraform templates managed by platform team post-deploy)   

---

## 4. Verification

I verified this change by:

- [x] Running automated tests
- [x] Running manual tests
- [x] Checking logs
- [ ] Checking metrics
- [ ] Testing error cases
- [ ] Testing permissions/security behavior
- [x] Testing rollback or failure behavior, if relevant

Commands run:

```bash
     # Helm lint — all 3 coder charts pass                                                                                                                     
     helm lint --strict charts/coder charts/coder-secrets charts/coder-db                                                                                      
                                                                                                                                                               
     # Helm template — renders 3 child Application CRs correctly                                                                                               
     helm template charts/coder --values charts/coder/ci/ci-values.yaml                                                                                        
                                                                                                                                                               
     # Local k3s testing                                                                                                                                       
     helm install coder oci://ghcr.io/coder/chart/coder --version 2.33.7 -n coder-test -f coder-app.yaml                                                       
     kubectl exec coder-db-1 -n coder-test -- psql -U postgres -d coder -c "SELECT count(*) FROM users;"                                                       
     # Result: 3 users (admin, beniejoypossi via OIDC, prebuilds)                                                                                              
                                                                   
```

Results:

```text
     # helm lint                                                                                                                            
     ==> Linting charts/coder                                                                                                                                  
     [INFO] Chart.yaml: icon is recommended                                                                                                                    
     ==> Linting charts/coder-secrets                                                                                                                          
     [INFO] Chart.yaml: icon is recommended                                                                                                                    
     ==> Linting charts/coder-db                                                                                                                               
     [INFO] Chart.yaml: icon is recommended                                                                                                                    
     3 chart(s) linted, 0 chart(s) failed                                                                                                                      
                                                                                                                                                               
     # Local test — OIDC login via Keycloak (auth.verif.fyi) confirmed working                                                                                 
     # Token contains "coder_roles": ["admin"]                                                                                                                 
     # User created with member role (Premium-gated IdP Sync)                                                                                                  
     # DB-assigned owner role persists through re-login  
```

---

## 5. Screenshots / Evidence

Add evidence here:

*      - Local k3s test: Coder v2.33.7 pod running, CNPG coder-db cluster healthy, OIDC login via auth.verif.fyi/                                                
       realms/camer-digital successful  

---

## 6. Risk Assessment

Risk level:

* [x] Low
* [ ] Medium
* [ ] High

Potential risks:

* CODER_OIDC_GROUP_MAPPING silently ignored on AGPL 

Mitigation:

*  documented in Keycloak  comment, roles assigned manually via DB/API.                                        
---

## 7. AI Usage Declaration

AI was used for:

* [x] Understanding existing code
* [x] Generating code
* [x] Refactoring
* [ ] Generating tests
* [x] Drafting documentation
* [ ] Reviewing the diff
* [ ] Not used

Human verification:

* [x] I understand every meaningful change in this PR
* [x] I checked generated code manually
* [ ] I checked generated tests manually
* [x] I removed unsupported AI assumptions
* [x] I accept responsibility for this PR

---

## 8. Reviewer Focus

Please focus your review on:

* [x] Correctness
* [x] Architecture
* [ ] Security
* [x] Performance
* [ ] Tests
* [x] Maintainability
* [ ] Product intent
* [ ] Edge cases

### CODE REVIEW

File: charts/coder-secrets/values.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  # NOTE: `{{ .password }}` is an ESO Go template variable (fetched from the
+  # `data` section below). The externalsecret.yaml template renders this with
+  # `| quote` (NOT `| tpl`), so Helm does NOT process the double braces — they
+  # pass through as a literal for ESO's v2 template engine. If this value is
+  # ever refactored to use `tpl` (e.g., to template the namespace), the braces
+  # MUST be double-escaped: `{{ "{{ .password }}" }}` (see lightbridge-db
+  # values.yaml for the tpl-brace pattern).
+  coder-db-url:
+    templateData:
+      url: "postgres://coder:{{ .password }}@coder-db-rw.coder.svc.cluster.local:5432/coder?sslmode=disable"
```
Comment (stephane-segning) [inline charts/coder-secrets/values.yaml]: **Correctness — DB password is interpolated raw into the connection DSN (not URL-encoded).** The URL is assembled as `postgres://coder:{{ .password }}@…`. If `coder_db_password` in ASM contains any URL-reserved character (`@ / : ? # % &` or a space), the DSN parses wrong and Coder fails to connect (or targets the wrong host/db) with no error pointing back at the password. lightbridge does not embed the raw password in a DSN this way, so this risk is new here. Constrain `coder_db_password` to `[A-Za-z0-9]` in ASM, or URL-encode it in the template.

File: charts/coder/templates/applications.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    # Values live in ai-helm-values (ADR-0056), read via the $values ref
+    # (Source C) — not inline.
+    - repoURL: {{ .Values.app.chart.repoURL | quote }}
+      chart: "."
+      targetRevision: {{ .Values.app.chart.targetRevision | quote }}
+      helm:
+        releaseName: {{ .Values.app.releaseName | quote }}
+        valueFiles:
+          - $values/environments/{{ $a.env | default "prod" }}/values/coder-app.yaml
+        ignoreMissingValueFiles: true
```
Comment (stephane-segning) [inline charts/coder/templates/applications.yaml]: **Deploy-correctness — `ignoreMissingValueFiles: true` silently degrades coder-app to upstream chart defaults.** Source A pulls workload values from `$values/environments/prod/values/coder-app.yaml` in the private ai-helm-values repo. If that file is not on ai-helm-values `main` **before** this PR merges, `ignoreMissingValueFiles` makes ArgoCD render the upstream Coder chart with NO `CODER_PG_CONNECTION_URL`/OIDC/ingress env — Coder crash-loops (the chart bundles no Postgres). This is the values-repo-first hazard CLAUDE.md calls out. Confirm the prod values file exists in ai-helm-values before merge.

File: charts/apps/values.yaml [hunk 3]
```diff
@@ -1093,11 +1093,23 @@ applications:
...
+  # --- Coder App-of-Apps orchestrator (ADR-0083, re-introduction) ---
+  # Renders three child Application CRs (coder-secrets wave 0, coder-db wave 1,
+  # coder-app wave 2) from charts/coder. As a control object (it emits
+  # Application CRs) this entry is controlPlane: true — the children themselves
+  # target home-remote. OCI mode (ADR-0055): orchestrator chart floats from OCI
+  # so its App-of-Apps children config deploys continuously. Children: secrets +
+  # db leaf charts from OCI, the app's upstream Coder chart (oci://ghcr.io/coder/
+  # chart/coder, pinned) + its cert overlay from ai-helm-values (see
+  # charts/coder values). Supersedes ADR-0027 removal.
+  - name: coder
```
Comment (stephane-segning) [inline charts/apps/values.yaml:1105]: **ADR-vs-implementation mismatch.** ADR-0083 states: *"LimitRange + ResourceQuota uncommented for the `coder` namespace."* But the `coder-defaults` LimitRange + `coder-quota` ResourceQuota in `extraObjects` (~line 1336) remain commented out, and this entry does not uncomment them. The `coder` namespace deploys with no resource guardrails — Coder-spawned workspace pods can consume the node unbounded. Either uncomment those blocks or correct the ADR.

File: charts/coder/ci/coder-app.yaml [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          app.kubernetes.io/name: coder
+          app.kubernetes.io/instance: {{ .Release.Name }}
+    - |
+      apiVersion: monitoring.coreos.com/v1
+      kind: ServiceMonitor
+      metadata:
+        name: coder
+        namespace: {{ .Release.Namespace }}
+        labels:
+          release: prometheus
```
Comment (stephane-segning) [inline charts/coder/ci/coder-app.yaml]: **Cleanup / inconsistency.** This ServiceMonitor carries `release: prometheus`, but the sibling `coder-db` PodMonitor (and lightbridge-db's PodMonitor) carry no such label — Alloy discovers monitors without a `release` filter here, so the label is dead. Harmless today, but if monitor discovery is ever scoped by `release`, coder-db metrics would silently drop while coder-app metrics survive — a hard-to-debug asymmetry. Drop the label for consistency with the DB PodMonitor.

File: charts/coder/ci/ci-values.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# CI lint fixture for the coder orchestrator chart.
```
Comment (stephane-segning) [inline charts/coder/ci/ci-values.yaml:1]: **Convention — these CI fixtures are never exercised by CI.** `charts/coder` has its own `templates/` dir and no `ai-helm.adorsys-gis.github.io/lint-mode: ci-values` annotation, so the helm-lint CI lints/renders it with the **default** `values.yaml`, not `ci/*-values.yaml`. Both `ci/ci-values.yaml` and `ci/coder-app.yaml` are therefore dead in CI (the PR's claimed `helm template … --values charts/coder/ci/ci-values.yaml` is not what CI runs), giving false confidence. Either add the `lint-mode: ci-values` annotation so the fixtures are used, or move `coder-app.yaml` out of `ci/` (it's a reference file, not a lint fixture).

File: charts/coder/values.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  chartName: coder-secrets   # OCI chart name
+  releaseName: coder-secrets
+  syncWave: "0"
+
+# ────────────────────────────────────────────────────────────────────────
+# coder-db child — the CloudNativePG database leaf chart (this repo).
+# Lower sync-wave than the app so Postgres is up before Coder starts.
+# ────────────────────────────────────────────────────────────────────────
+db:
+  enabled: true
```
Comment (stephane-segning) [inline charts/coder/values.yaml]: Have this disabled by default. We won't use it in prod. As mentioned [here](https://github.com/ADORSYS-GIS/ai-helm-values/pull/85#pullrequestreview-4701448798), we will use the same cluster as LightBridge and create a custom DB for Coder there.

File: charts/coder-db/values.yaml
Comment (stephane-segning) [inline charts/coder-db/values.yaml:1]: This chart `coder-db` has no reason of existance. In fact, it's just a passthrough via `tpl` in resources.yaml.

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Review by Koufan-De-King — APPROVED** [review]

**Review by stephane-segning — APPROVED** [review]
